sdk/src/main/java/com/meniga/sdk/webservices/PersistenceDelegate.java

```diff
@@ -474,38 +475,50 @@ public void onFailure(Call<E> call, Throwable t) {
 		if (provider.hasKey(req)) {
 			return createTask(provider.fetch(req));
 		}
-		return persist(req, getClient(Service.USERS).getUserProfile());
+		return persist(req, getService(UsersService.class).getUserProfile());
 	}
 
 	public Result<List<MenigaUser>> getUsers(GetUsers req) {
 		if (provider.hasKey(req)) {
 			return createTask(provider.fetch(req));
 		}
-		return persist(req, getClient(Service.USERS).getUsers());
+		return persist(req, getService(UsersService.class).getUsers());
 	}
 
 	public Result<Void> setCulture(SetCulture req) {
-		return persist(req, getClient(Service.USERS).setCulture(req.toQueryMap()));
+		return persist(req, getService(UsersService.class).setCulture(req.toQueryMap()));
 	}
 
 	public Result<MenigaUser> registerUser(RegisterUser req) {
-		return persist(req, getClient(Service.USERS).registerUser(req));
+		return persist(req, getService(UsersService.class).registerUser(req));
 	}
 
 	public Result<Void> forgotPassword(ForgotPassword req) {
-		return persist(req, getClient(Service.USERS).forgotPassword(req));
+		return persist(req, getService(UsersService.class).forgotPassword(req));
 	}
 
 	public Result<List<MenigaUserMetaData>> getUserMetaData(GetUserMetaData req) {
-		return persist(req, getClient(Service.USERS).getUserMetaData(req.toQueryMap()));
+		return persist(req, getService(UsersService.class).getUserMetaData(req.toQueryMap()));
 	}
 
 	public Result<MenigaUserMetaData> saveUserMetaData(SaveMetaData req) {
-		return persist(req, getClient(Service.USERS).saveUserMetaData(req));
+		return persist(req, getService(UsersService.class).saveUserMetaData(req));
 	}
 
 	public Result<Void> resetPassword(ResetPasswordWithToken req) {
-		return persist(req, getClient(Service.USERS).resetPassword(req));
+		return persist(req, getService(UsersService.class).resetPassword(req));
+	}
+
+	public Result<Void> updateEmail(UpdateEmail req) {
+		return call(getService(UsersService.class).updateEmail(req));
```

Comment:
Should call persist so that clients get notified to invalidate their cache when this happens, probably same for deleteUser and changePassword. Can think about renaming "persist".

Reply:
I can do so for `updateEmail` and `changePassword` but there's no key that I could depend on in case of `deleteUser`.

Reply:
The requests objects were meant to be the key - even if there's no data to pass along with it. I guess we can omit it for this request. This operation must log the user out of the client app anyway which is beyond the scope of the sdk.